R/functions_netCDF.R

```diff
@@ -337,105 +424,248 @@ create_netCDF_from_array_with_variables <- function(x, locations, grid,
+    var_start <- c(var_start, 1)
+  }
 
-  if (has_time_central) {
-    var_chunksizes <- c(var_chunksizes, 1L)
+  if (has_Z_verticalAxis) {
+    var_chunksizes <- c(var_chunksizes, z_chunksize)
     var_start <- c(var_start, 1)
   }
 
-  # define dimensions
+  # define dimensions ---------------------------------------------------------------------
+
+  #  bounds dimension
   bnddim <- ncdf4::ncdim_def(name = "bnds", units = "", vals = seq_len(2L),
-    create_dimvar = FALSE)
-  xdim <- ncdf4::ncdim_def(name = "lon", longname = "Longitude",
-    units = "degrees_east", vals = xvals)
-  ydim <- ncdf4::ncdim_def(name = "lat", longname = "Latitude",
-    units = "degrees_north", vals = yvals)
-  if (has_time_central) {
-    tdim <- ncdf4::ncdim_def(name = "time", units = "Gregorian_year since 1900",
-      vals = time_central)
-  }
-
-  # define variables
-  var_dims <- list(xdim, ydim)
-  if (has_time_central) {
+                                           create_dimvar = FALSE)
+
+  # lat and long dimension
+  if(isGridded){
+    xdim <- ncdf4::ncdim_def(name = "lon", longname = "Longitude",
+                             units = "degrees_east", vals = xvals)
+    ydim <- ncdf4::ncdim_def(name = "lat", longname = "Latitude",
+                             units = "degrees_north", vals = yvals)
+  } else {
+    idim <- ncdf4::ncdim_def(name = 'site', longname = 'SOILWAT2 simulation sites',
+                             units = 'site_id', vals = 1:nrow(data))
+  }
+
+  # time dimension
+  if(has_T_timeAxis) {
+      tdim <-  ncdf4::ncdim_def(name = time_names,
+                                units = time_units,
+                                calendar = time_cal,
+                                vals = time_vals)
+  }
+
+  # vertical dimension
+  if(has_Z_verticalAxis) {
+    zdim <-  ncdf4::ncdim_def(name = vert_names,
+                              units = vert_units,
+                              vals = vert_vals)
+  }
+
+  # define dimensionality of netcdf variables -------------------------------------------------
+  var_dims <- if(isGridded) list(xdim, ydim) else list(idim)
+
+  if (has_T_timeAxis) {
     var_dims <- c(var_dims, list(tdim))
   }
 
-  var_defs <- lapply(seq_len(nl), function(k)
+  if (has_Z_verticalAxis) {
+    var_dims <- c(var_dims, list(zdim))
+  }
+
+  if(length(time_vals) > 1 || length(vert_vals) > 1) nn <- 1 else nn <- nl
+
+  var_defs <- lapply(seq_len(nn), function(k)
     ncdf4::ncvar_def(name = var_names[k], units = var_units[k],
       dim = var_dims, chunksizes = var_chunksizes, missval = NAflag,
       longname = var_longnames[k], prec = ncdf4_datatype))
 
+  # add lat and long as variables if not gridded
+  if(!isGridded){
+   latvar <-  ncdf4::ncvar_def(name = 'lat', units = 'degrees_north',
+                               dim = list(idim), chunksizes = var_chunksizes[1], missval = NAflag,
+                               longname = 'site latitiude', prec = ncdf4_datatype)
```

Comment:
line 515: longname is misspelled
line 515 and 519: why do we add "site" to the lat/long longnames?